### ISSUE TITLE
Referencing formula numbers in articles

### ISSUE THREAD

**Issue by falgon** [falgon/roki-web#34]
The expression number is inserted directly in the article, but it should be expressed by `\label`, `\ref`, `\eqref`, etc.
However, [KaTeX (0.12.0) does not currently support them](https://katex.org/docs/support_table.html#e), so this is not feasible. If this is implemented, we plan to introduce it sequentially. (This was possible because the old blog used mathjax).

* related: 
    * https://github.com/KaTeX/KaTeX/issues/350
    * https://github.com/KaTeX/KaTeX/issues/2003

**Comment by zhangpeng96** [comment]
Thanks to your solution. I use katex's marco function and wrap it up with elegant code.
KaTeX's configuration:
```javascript
trust: ["\\htmlId"],
macros: {
  "\\eqref": "\\href{###1}{(\\text{#1})}",
  "\\ref": "\\href{###1}{\\text{#1}}",
  "\\label": "\\htmlId{#1}{}"
}
```
Then you can use `\eqref` & `\ref` & `\label` command same to latex or mathjax.

more details: https://column.mowchan.me/post/2021/katex-ref-label-support.html

**Comment by LXY631** [comment]
你好，不知有时间可以写一个vs code插件markdowm+math（不是markdown math）的配置文件吗，资料看了半天没有搞好，谢谢了。
软件预设的.json文件模板如下：
 "mdmath.macros": {
      
    }

**Comment by LXY631** [comment]
@zhangpeng96 你好，不知有时间可以写一个 vs code 插件 markdowm+math（不是 markdown math）的配置文件吗，资料看了半天没有搞好，谢谢了。
软件预设的.json 文件模板如下：
"mdmath.macros": {

}

**Comment by zhangpeng96** [comment]
> @zhangpeng96 你好，不知有时间可以写一个 vs code 插件 markdowm+math（不是 markdown math）的配置文件吗，资料看了半天没有搞好，谢谢了。
> 软件预设的.json 文件模板如下：
> "mdmath.macros": {
> 
> }

markdowm是什么？是markdown打错了吗？关于katex的配置可以看官方的文档：https://github.com/KaTeX/KaTeX/blob/master/docs/options.md

vscode插件编写我不了解，没办法帮你

**Comment by hyliang96** [comment]
> @zhangpeng96 你好，不知有时间可以写一个 vs code 插件 markdowm+math（不是 markdown math）的配置文件吗，资料看了半天没有搞好，谢谢了。 软件预设的.json 文件模板如下： "mdmath.macros": {
> 
> }

安装vscode插件: [[Markdown+Math](https://github.com/goessner/mdmath)]，并在vscode配置文件里添加如下设置，即可渲染\label、\ref、\eqref

~~~json
"mdmath.katexoptions": {
    "trust": "(context) => ['\\htmlId', '\\href'].includes(context.command)"
},
"mdmath.macros": {
    "\\eqref": "\\href{###1}{(\\text{#1})}",
    "\\ref": "\\href{###1}{\\text{#1}}",
    "\\label": "\\htmlId{#1}{}"
},
~~~
